dataclasses_avroschema/avrodantic.py

```diff
@@ -29,10 +29,12 @@ def asdict(self, standardize_factory: Optional[Callable[..., Any]] = None) -> Js
         """
         data = dict(self)
 
+        standardize_method = standardize_factory or standardize_custom_type
+
         # te standardize called can be replaced if we have a custom implementation of asdict
         # for now I think is better to use the native implementation
         return {
-            key: value.asdict() if isinstance(value, AvroBaseModel) else standardize_custom_type(value)
+            key: value.asdict() if isinstance(value, AvroBaseModel) else standardize_method(value)
             for key, value in data.items()
         }
```

Comment:
Since `standardize_custom_type` will handle pydantic.BaseModel with `is_pydantic_model(type(value))`, then this `asdict` seems unecessary and this part could be rewritten as
```
```suggestion
            return standardize_method(data)
```

Reply:
makes sense - let's figure out `dict` vs. `asdict` discussion first, later I'll change it.

Reply:
done